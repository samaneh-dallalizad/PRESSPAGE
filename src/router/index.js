import { createRouter, createWebHistory } from 'vue-router'
import HomePage from './../components/HomePage/index.vue'
import Users from './../components/Users/index.vue'

const routes = [

    {
        path: '/',
        name: 'Home',
        component: HomePage
    },
    {
        path: '/users',
        name: 'users',
        component: Users,
    }
]
const router = createRouter({
    history: createWebHistory(),
    routes
})
export default router
