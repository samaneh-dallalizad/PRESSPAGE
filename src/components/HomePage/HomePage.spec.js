import Vue from 'vue';
import { mount } from '@vue/test-utils'
import HomepageComponent from './index.vue';
import {it} from 'vitest';
import router from "../../router"

describe('HomepageComponent', () => {
  it('homepage must always be the same', async () => {
    router.push("/")
    await router.isReady()
    const wrapper = mount(HomepageComponent, {
      global: {
        plugins: [router]
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})