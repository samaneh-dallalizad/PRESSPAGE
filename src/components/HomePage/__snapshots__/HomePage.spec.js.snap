// Vitest Snapshot v1

exports[`HomepageComponent > homepage must always be the same 1`] = `
<section
  class="home-page"
  data-v-7ce3c19b=""
>
  <div
    class="container mx-auto"
    data-v-7ce3c19b=""
  >
    <div
      class="relative mx-auto inline-block"
      data-v-7ce3c19b=""
    >
      <span
        class="flex text-center items-center bg-pink-500 rounded-full w-60 h-60 justify-center"
        data-v-7ce3c19b=""
      >
        <a
          class="inline-block w-40 h-40 p-4 text-white rounded-full btn--jump"
          data-v-7ce3c19b=""
          href="/users"
        >
          <svg
            aria-hidden="true"
            class="w-20 h-20 mx-auto align-middle mt-5"
            data-v-7ce3c19b=""
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 20h5v-2a3 3 0 00-5.356-1.857M17 20H7m10 0v-2c0-.656-.126-1.283-.356-1.857M7 20H2v-2a3 3 0 015.356-1.857M7 20v-2c0-.656.126-1.283.356-1.857m0 0a5.002 5.002 0 019.288 0M15 7a3 3 0 11-6 0 3 3 0 016 0zm6 3a2 2 0 11-4 0 2 2 0 014 0zM7 10a2 2 0 11-4 0 2 2 0 014 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </a>
      </span>
    </div>
  </div>
</section>
`;
